moby/moby
# fix slow response for log read request with --since parameter for jsonfilelog

Signed-off-by: rpei <cspeiyb@gmail.com>



**- What I did**
Fix the issue as described in #31935 
**- How I did it**
  1. set --log-opt max-size to a small size, for example, 5m

  2. Based on the real need, set --log-opt max-file to one reasonable number.

  3. Before building the logic concatenation ReadSeeker, make code change to only select files whose mtime is later than the --since parameter. This could help reduce the range of total log records and let the docker daemon to give a quick response to client.
**- How to verify it**
1.Run a container to generate enough logs, the more, the better. In my test, the total log size is around 400M
2.Using the following command to verify how long to get a response from docker daemon. Please specify a relatively late timestamp to the since parameter. For a extreme case, you could specify a latest timestamp of all the logs to the --since parameter.
date;docker logs --since=xxxxx [container_id] > test.log 2>&1;date
**- Description for the changelog**
Make enhancement to reduce the total size of logs to search for the log read request with --since parameter.


**- A picture of a cute animal (not mandatory but encouraged)**
